src/emscatter.jl

```diff
@@ -0,0 +1,296 @@
+# In this tutorial, we will learn:
+# 
+#   * How to formulate the weak form for time-harmonic electromagnetic problem
+#   * How to implement perfectly matched layers (PML)
+#   * How does periodic boundary condition work in Gridap
+# 
+# # Problem statement
+# 
+# The problem that we are trying to solve is the plane wave (TM-polarized $H_{inc}$) scattering of a cylinder dielectric (of radius $R$ and permittivity $\varepsilon$), as illustrated below. The computational cell is of height $H$ and length $L$, and we designate a perfectly matched layer (PML) thickness of $d_{pml}$.
+# 
+# ![](../assets/emscatter/Illustration.png)
+# 
+# The governing equation of this scaler electromagnetic problem in 2D is the Helmholtz equation: 
+#
+# ```math
+# \left[-\nabla\cdot\frac{1}{\varepsilon(x)}\cdot\nabla -k^2\mu(x)\right] H = f(x),
+# ```
+#
+# where $k=\omega/c$ is the wave number in free space and $f(x)$ is the source term
```

Comment:
source term (which corresponds to a magnetic current density in Maxwell's equations).